### PR TITLE
fix: pages revamping

### PR DESCRIPTION
**Pull request by NarayanBavisetti**
### **Pages**
In this release, we've restructured pages significantly. Previously, users created pages with distinct blocks, utilizing a title and description format. However, we've now streamlined this process by eliminating the concept of page blocks. Upon creating a page, users now have access to a comprehensive text editor, empowering them to create content with diverse elements and enhanced flexibility.

### Key Updates:

- Archiving and Unarchiving: Users can now archive and subsequently unarchive pages.
- Page Locking: A new feature allows the locking and unlocking of pages, restricting editing access for specific users.
- Page Linking: Users can establish links between existing pages, facilitating seamless navigation within the system.
- Child Page Creation: Pages can now serve as parent entities, enabling the creation of child pages, fostering a hierarchical structure for organized content management.

### CODE REVIEW

File: apiserver/plane/api/serializers/page.py [hunk 1]
```diff
@@ -102,6 +80,41 @@ def update(self, instance, validated_data):
...
+            "page",
+        ]
+
+    def get_entity_details(self, obj):
+        entity_name = obj.entity_name
+        if entity_name == 'forward_link' or entity_name == 'back_link':
+            try:
+                page = Page.objects.get(pk=obj.entity_identifier)
+                return PageSerializer(page).data
+            except Page.name:
```
Comment (pablohashescobar) [inline apiserver/plane/api/serializers/page.py:101]: @NarayanBavisetti can you check this this should be `Page.DoesNotExist`

File: apiserver/plane/db/migrations/0049_auto_20231116_0713.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    try:
+        Page = apps.get_model("db", "Page")
+        PageBlock = apps.get_model("db", "PageBlock")
+        PageLog = apps.get_model("db", "PageLog")
+
+        updated_pages = []
+        page_logs = []
+
+        # looping through all the pages
+        for page in Page.objects.all():
```
Comment (pablohashescobar) [inline apiserver/plane/db/migrations/0049_auto_20231116_0713.py:19]: use `values_list`

**Review by pablohashescobar — APPROVED** [review]

File: apiserver/plane/api/serializers/page.py (hunk 1 above)
Comment (NarayanBavisetti) [reply apiserver/plane/api/serializers/page.py]: Yes. Made the change

File: apiserver/plane/db/migrations/0049_auto_20231116_0713.py (hunk 2 above)
Comment (NarayanBavisetti) [reply apiserver/plane/db/migrations/0049_auto_20231116_0713.py:19]: You can't directly update dictionaries or tuples obtained from values() or values_list() and then perform a bulk update.

**Review by sriramveeraghanta — APPROVED** [review]

**Pull request merged**